### DISCUSSION TITLE
Slingbox SB260 Solo Performance Issue on Raspberry Pi....

### DISCUSSION THREAD

**Discussion by highveld66** [GerryDazoo/Slinger#139] [Q&A]
I have tried SB350 successfully. Work with both windows and raspberry pi
But when I try a SB260 the windows server works great
But the raspberry pi B+  is very laggy , stutters....
Has anyone come across this
The same raspberry Pi & SD card buld works great for the SB350.
So it points to SB250 and or config... 

Any ideas would be welcome

Thanks

**Comment by GerryDazoo** [comment]
As far as I know my code has never been tested with a SB250. We did get a 240 running. What SBtype are you using? in you config.ini. What version of the code are you running.

**Reply by highveld66** [reply]
Hello , wanted to say a big thanks for you work on this.  Really nice soluion. On you question on type , i have used both type definitions under tests. Surpising the SB350 one works slighty better! even though its a SB260   I also tried reducing qulity etc but it did have a major benefit. Feel like a subtle issue, unfortunately the worse to find ;-)  I am running the older versionpre your latest build. I note a guy relied he has tried both builds with similar issues.....  I suspected singbox hardware not being upto it. but when it worked on windows that scrapped that theory!

**Comment by jkaponis** [comment]
Hello. I have the same problem With a pro hd, works perfect on windows server but on the raspberry pi B+ is laggy, with the hd pro the load on the raspberry cpu is 100% and with the 350 only 12% . my 350 works perfect on both of them with version 2.0 and 3.8. any ideas what can be wrong with my configuration.  thanks

**Reply by highveld66** [reply]
Snap then! Lets hope we can find out why.... At least we have our SB350's

**Reply by GerryDazoo** [reply]
Are you running this over WiFi? That's where I would start looking assuming you've make all the required TCP buffer size mods as documented in the readme.

**Reply by GerryDazoo** [reply]
Also, the CPU show never be 100% just running my code. Basically once the stream is setup it's just a packet in/packet out loop. Are you using "top" to display the CPU? What else is running. There is a little more per packet packet processing for the Solo but in my testing it's never been significant.

**Reply by jkaponis** [reply]
Hello. Thank for your reply. I use htop. the raspberry py only run slinger over utp on the same switch. The 350 load is max id 12% but if i use the prohd htop say 100% and 2.6 load. I follow all the readme instructions. Both of the slingbox works perfect on Windows wiht similar load of cpu. Thanks

**Reply by highveld66** [reply]
Hello Gerry, No running wired to switch. Then wifi for client on internal lan. Infra is same between working SB350 and stuttering SB260
If it was not for the two cases one good and one bad then i would look to tin and wire more. i will have a look at perf counters on the pi for the two cases. and let you know. 
for clarity its on a bare build no gui and nothing else running on pi

 thanks for helping will get back to you

**Comment by GerryDazoo** [comment]
Try setting Resolution to 5 and VideoBandwidth to 1000. Let's see if that makes a difference.

**Reply by highveld66** [reply]
Hello Gerry tried the config change and seems to be the same ie no improvement laggy / breakup 
One other things is I think the remote is not wrking much, 
Very ocassionlly I can get a power off with the test button. But again same kit / infra works fine on SM350?
I will do some perf counters next and let you know  Thanks again

**Reply by GerryDazoo** [reply]
please supply your config.ini and all the program output

**Reply by GerryDazoo** [reply]
My Raspberry Pi is a thousand miles away in a drawer right now but I do have another Linux box I turned up. Here's the scoop, when using a 350/500 the code just grabs a packet and sends a packet, very low overhead BUT the Solo/Pro  etc actually has portions of the video stream encrypted so that's where your CPU cycles are going. The thread that's processing the Solo video stream is pegging the CPU. 
So all I can say is it is what it is BUT I'd be very interested if you could try running it with "PyPy" and see what happens. I've used PyPy in the past and have seen 10x performance improvements.

- Gerry

**Reply by highveld66** [reply]
Will have a go at this Gerry , thanks for suggestion. Will also try it on a pi 3a+ which is more powerful..... Makes perfect sense that more processing is causing this lag. Good spot. Did you sniff the packets to work this out? Just curious. 
Thanks again

**Reply by highveld66** [reply]
Tried the pi 3a+, and this still has issues with Lag and Stutter
Will have a look at pypy next
Thought it usefull to say what I found so far
